client/src/routes/Sessions/components/JoinSessionModal.tsx

```diff
@@ -33,10 +33,15 @@ const JoinSessionModal = () => {
         <VerificationCode
           prefillCode={`${inviteCode || ''}`}
           onCodeCompleted={async value => {
-            const session = await joinSession(value);
-            fetchSessions();
-            goBack();
-            navigate('SessionModal', {session: session});
+            try {
+              const session = await joinSession(value);
+              fetchSessions();
+              goBack();
+              navigate('SessionModal', {session: session});
+            } catch (err) {
+              goBack();
+              navigate('SessionUnavailable');
+            }
           }}
         />
```

Comment:
Outside of this PR, this modal needs better error handling. There's a [todo about that in notion](https://www.notion.so/29k/Closed-BETA-0055aabd797b40c7b390b7cea4eba658#843cd87bd79248fe91898095d737f0c8). You can also have a look at how it's solved in host invites https://github.com/29ki/29k/blob/main/client/src/routes/Profile/UpgradeAccount.tsx#L147-L162

Reply:
Not sure if I follow, this is that flow, the difference I see is  that it navigates to a new modal that is specific for that user-flow, rather than render something different in the same modal, is that what you meant by better error handling? 🤔

Reply:
So if you write the wrong code manually you will not be able to edit it, you will be thrown to this modal instead? Do I understand this correctly? 
In that case it feels like two different cases. One is dealing with bad user input or general errors in `JoinSessionModal`. The other one is dealing with a session that is old/ended/deleted. 
Did you see my comment here? https://github.com/29ki/29k/pull/871#pullrequestreview-1168444119

Reply:
Ah didn't see that comment, thanks!

The two other cases you mentioned seems super rare to me but maybe it would happen more often than I imagine.

Good point, I just don't see how handling it on `useSubscribeToSession` would help on the flow that is being handled atm.

If you only have a invite code or clicked a link, we don't really use `useSubscribeToSession` right? unless we subscribe to session by invite code or something....

And the endpoint to join the session just fails, it doesn't say why (because no session was found with that token or because session is not available any longer)

We could return different error codes to handle them individually... Can sync with Lisa again about it since we discussed it these days.

Reply:
but handling it on `useSubscribeToSession` would definitely work for the two cases you mentioned 👌

Reply:
Great! So in this PR let's forget about the `useSubscribeToSession` error.

Could you adjust this so instead of being a separate modal, the error is shown where you input the code. And then we deal with old/ended/deleted sessions outside of this one? :)

Reply:
I guess I need to sync with Lisa about it, we agreed on showing a modal just like the SessionModal communicating that the session is not available.

And we still have the two cases, "misstype/invalid invite code" (session not found) and session not available anymore

Reply:
Would the error you mentioned still show the input to type the code again?

Reply:
Sounds good :) I think we could say that session not found and not available is the same thing. But I leave that up to you and Lisa.

Reply:
And if you implement two different errors, please align with how host requests are handled in `UpgradeAccount` :) 
https://github.com/29ki/29k/blob/main/client/src/routes/Profile/UpgradeAccount.tsx#L70-L95